src/app/shared/services/dia-backend/asset/uploading/dia-backend-asset-uploading.service.ts

```diff
@@ -31,26 +32,31 @@ import { DiaBackendAssetRepository } from '../dia-backend-asset-repository.servi
   providedIn: 'root',
 })
 export class DiaBackendAssetUploadingService {
-  private readonly _isPaused$ = new BehaviorSubject(false);
-  private readonly _isPausedByFailure$ = new BehaviorSubject(false);
+  private readonly preferences = this.preferenceManager.getPreferences(
+    DiaBackendAssetUploadingService.name
+  );
   private readonly _taskQueue$ = new BehaviorSubject<Proof[]>([]);
-  private readonly _currentUploadingCount$ = new BehaviorSubject(0);
-  readonly isPaused$ = this._isPaused$
-    .asObservable()
-    .pipe(distinctUntilChanged());
-  readonly isPausedByFailure$ = this._isPausedByFailure$
+  // tslint:disable-next-line: rxjs-no-explicit-generics
+  private readonly _pendingTasks$ = new BehaviorSubject<number | undefined>(
+    undefined
+  );
+  readonly isPaused$ = this.preferences.getBoolean$(PrefKeys.IS_PAUSED);
+  readonly networkConnected$ = this.networkService.connected$;
+  private readonly executionSignal$ = combineLatest([
```

Comment:
I think we should use `event` instead of `signal` for naming consistentcy in this project.

```suggestion
  private readonly executionEvent$ = combineLatest([
```